src/crafting.h

```diff
@@ -104,6 +104,84 @@ struct recipe {
 
 };
 
+/**
+*   enum used by comp_selection to indicate where a component should be consumed from.
+*/
+enum usage {
+    use_from_map = 1,
+    use_from_player = 2,
+    use_from_both = 1 | 2,
+    use_from_none = 4,
+    cancel = 5 // FIXME: hacky.
```

Comment:
There's 4 cases that need changing in crafting.cpp at lines 2038, 2049, 2161 and 2164.